server/plugins/pfed-sso/src/main/java/com/walmartlabs/concord/server/plugins/pfedsso/SsoClient.java

```diff
@@ -0,0 +1,215 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =====
+ */
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.apache.http.HttpHeaders;
+import org.immutables.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+@Named
+public class SsoClient {
+    private static final Logger log = LoggerFactory.getLogger(SsoClient.class);
+
+    private static final String USER_AGENT_HEADER = "Mozilla/5.0";
+    private static final String ACCEPT_LANG_HEADER = "en-US,en;q=0.5";
+    private static final String CONTENT_TYPE_HEADER = "application/x-www-form-urlencoded";
+    private static final String CHARSET_HEADER = "utf-8";
+    private static final String TOKEN_REQUEST = "code=%s&redirect_uri=%s&grant_type=authorization_code&client_id=%s";
+    private static final String REVOKE_TOKEN_REQUEST = "token=%s&token_type_hint=refresh_token&client_id=%s";
+
+    private final SsoConfiguration cfg;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Inject
+    public SsoClient(SsoConfiguration cfg) {
+        this.cfg = cfg;
+        try {
+            TrustManager[] trustAllCerts = new TrustManager[]{
+                    new X509TrustManager() {
+                        public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                            return null;
+                        }
+
+                        public void checkClientTrusted(
+                                java.security.cert.X509Certificate[] certs, String authType) {
+                        }
+
+                        public void checkServerTrusted(
+                                java.security.cert.X509Certificate[] certs, String authType) {
+                        }
+                    }
+            };
+
+            SSLContext sc = SSLContext.getInstance("SSL");
+            sc.init(null, trustAllCerts, new java.security.SecureRandom());
+            HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
```

Comment:
This looks like a global setting. We probably should not be doing this as it might affect other libraries and/or server plugins.

Reply:
removed as image has trusted cert in it